app/models/approval.rb

```diff
@@ -53,14 +54,14 @@ def approved_at
     end
   end
 
-  # Used by the state machine
-  def on_rejected_entry(new_state, event)
-    self.proposal.reject!
-  end
-
-  # Used by the state machine
-  def on_approved_entry(new_state, event)
-    self.proposal.partial_approve!
-    Dispatcher.on_approval_approved(self)
+  # notify parents if we've been approved. Notified as a callback so that it
+  # will be present even if subclasses override workflow
+  def on_approved_entry(old_state, event)
+    if self.parent
+      self.parent.child_approved!
+    else
+      self.proposal.approve!
+    end
+    self.reload   # Account for proposal changes
```

Comment:
This seems like it shouldn't be necessary in here...more the responsibility of the caller.